internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -460,19 +405,18 @@ public CompletableFuture<VeniceResourceCloseResult> closeAsync(
       if (isClosed) {
         return CompletableFuture.completedFuture(VeniceResourceCloseResult.ALREADY_CLOSED);
       }
-    }
 
-    return CompletableFuture.supplyAsync(() -> {
-      synchronized (closeLock) {
-        if (isClosed) {
-          return VeniceResourceCloseResult.ALREADY_CLOSED;
-        }
```

Comment:
I think the following check is still there in the code
 `if (isClosed) {
        return CompletableFuture.completedFuture(VeniceResourceCloseResult.ALREADY_CLOSED);
      }
```

Reply:
Right, but we only set `isClosed = true` when the segments and producer are fully closed in the task thread asynchronously. If there are concurrent closeAsync arrived, then sounds like it is possible that we can have multiple tasks to close the same producer.

Reply:
I think we should move back the following check inside ` CompletableFuture.supplyAsync` block and remove outer check 

```
        if (isClosed) {
          return VeniceResourceCloseResult.ALREADY_CLOSED;
        }
```

Reply:
I think we should move back `isClosed` check inside ` CompletableFuture.supplyAsync` block and remove the outer one. 

For RejectedExecutionException I think we can put it inside try-catch block and swallow the exception as double close is not expected anyway

Reply:
Sure, I don't mind adding back the `if (isClosed)` check inside the closure. I think we do need to keep the outer one, though, because we want to avoid submitting a future into a shutdown `threadPoolExecutor`.